src/analyses/goto_check.cpp

```diff
@@ -1082,6 +1077,18 @@ void goto_checkt::pointer_validity_check(
         expr,
         guard);
     }
+
+    if(flags.is_unknown() ||
+       flags.is_integer_address())
+    {
+      add_guarded_claim(
+        implies_exprt(integer_address(pointer), allocs),
+        "dereference failure: invalid integer address",
+        "pointer dereference",
+        expr.find_source_location(),
+        expr,
+        guard);
+    }
```

Comment:
I suppose this change and what I mourned above just need to be moved into a separate commit.